doc/deployment/on_premises.md

```diff
@@ -1,33 +1,181 @@
+Persistent volumes are provisioned in Kubernetes like all things in Kubernetes: by means of a manifest.
+You can learn about creating [volumes](https://kubernetes.io/docs/concepts/storage/volumes/)  and [persistent volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) in the Kubernetes documentation.
+
+You or your Kubernetes administrators will be responsible for configuring the PVs you create to be consumable as static PV's, with Persistent Volume Provisioning or as a StorageClass.
+
+#### What you'll need for Pachyderm configuration of PV storage
+
+Keep the information below at hand for when you [run `pachctl deploy custom` further on](./deploy_custom.html)
+
+##### Configuring with static volumes
+
+You'll need the name of the PV and the amount of space you can use, in gigabytes.
+We'll refer to those, respectively, as `PVC_STORAGE_NAME` and `PVC_STORAGE_SIZE` further on.
+With this kind of PV, 
+you'll use the flag `--static-etcd-volume` with `PVC_STORAGE_NAME` as its argument in your deployment.
+
+Note: this will override any attempt to configure with StorageClasses, below.
+
+##### Configuring with StatefulSets
+
+If you're deploying using [StatefulSets](#statefulsets),
+you'll just need the amount of space you can use, in gigabytes, 
+which we'll refer to as `PVC_STORAGE_SIZE` further on..
+
+Note: The `--etcd-storage-class` flag and argument will be ignored if you use the flag `--static-etcd-volume` along with it.
+
+##### Configuring with StatefulSets using StorageClasses
+
+If you're deploying using [StatefulSets](#statefulsets) with [StorageClasses](#storageclasses), 
+you'll need the name of the storage class and the amount of space you can use, in gigabytes.
+We'll refer to those, respectively, as `PVC_STORAGECLASS` and `PVC_STORAGE_SIZE` further on.
+With this kind of PV,
+you'll use the flag `--etcd-storage-class` with `PVC_STORAGECLASS` as its argument in your deployment. 
+
+Note: The `--etcd-storage-class` flag and argument will be ignored if you use the flag `--static-etcd-volume` along with it.
+
+   
+### Deploying an object store
+
+#### Object store: what's it for?
+An object store is used by Pachyderm's `pachd` for storing all your data. 
+The object store you use must be accessible via a low-latency, high-bandwidth connection like [Gigabit](https://en.wikipedia.org/wiki/Gigabit_Ethernet)  or [10G Ethernet](https://en.wikipedia.org/wiki/10_Gigabit_Ethernet).
+
+For an on-premises deployment, 
+it's not advisable to use a cloud-based storage mechanism.
+Don't deploy an on-premises Pachyderm cluster against cloud-based object stores such as S3 from [AWS](https://pachyderm.readthedocs.io/en/latest/deployment/amazon_web_services.html), GCS from [Google Cloud Platform](https://pachyderm.readthedocs.io/en/latest/deployment/google_cloud_platform.html), Azure Blob Storage from  [Azure](https://pachyderm.readthedocs.io/en/latest/deployment/azure.html).
```

Comment:
Maybe clarify why not? Performance?

Reply:
I could clarify. It's a little clearer further down where I specify the performance requirements for on-prem storage deployments (gig or 10G ethernet or better).  

There are hybrid cloud environments where this is feasible, but the reason it's feasible is that there's a lot of peering-style networking going on, along with on-prem caching.